Subscriber/ApplePayDirectSubscriber.php

```diff
@@ -115,6 +194,55 @@ public function onFrontendCheckoutPostDispatch(Enlight_Event_EventArgs $args)
         $view->assign('sPayments', $sPayments);
     }
 
+    /**
+     * @param int $customerId
+     * @param int $paymentId
+     * @throws \Doctrine\ORM\ORMException
+     * @throws \Doctrine\ORM\OptimisticLockException
+     */
+    private function updateCustomerDefaultPaymentMethod(int $customerId, int $paymentId) {
+        $repository = $this->modelManager->getRepository(Customer::class);
+        $customer = $repository->find($customerId);
+
+        $customer->setPaymentId($paymentId);
+
+        $this->modelManager->persist($customer);
+        $this->modelManager->flush();
+    }
+
+    /**
+     * @param int $customerId
+     * @return mixed|null
+     * @throws \Doctrine\ORM\NoResultException
+     * @throws \Doctrine\ORM\NonUniqueResultException
+     */
+    private function getCustomerDefaultNonApplePayPaymentMethod(int $customerId) {
+        $paymentId = $this->config->get('defaultpayment');
+
+        $queryBuilder = $this->modelManager->createQueryBuilder();
+        $queryBuilder
+            ->select('orders.paymentId')
+            ->from(Order::class, 'orders')
+            ->leftJoin('orders.payment', 'payment')
+            ->where('orders.customerId = :customerId')
+            ->andWhere('payment.name != :applePayName')
+            ->andWhere('payment.name != :applePayDirectName')
+            ->orderBy('orders.id', 'DESC')
+            ->setParameter('customerId', $customerId)
+            ->setParameter('applePayName', self::APPLEPAY_NAME)
+            ->setParameter('applePayDirectName', self::APPLEPAYDIRECT_NAME)
+            ->setMaxResults(1)
+        ;
+
+        $customerOrders = $queryBuilder->getQuery()->getSingleResult();
```

Comment:
what happens if the customer has no order? doesnt getSingleResult imply a single result?
so i guess there is an exception and the if from below isnt triggered in that case?

Reply:
Fixed

Reply:
static code analysis would prevent those issues